internal/upgradecheck/http_test.go

```diff
@@ -144,22 +150,28 @@ func TestCheckForUpgrades(t *testing.T) {
 				}, nil
 			}
 			counter++
-			json := `{"display_name":"PGO 5.0.3","major_version":5}`
+			json := `{"pgo_versions":[{"tag":"v5.0.3"},{"tag":"v5.0.2"},{"tag":"v5.0.1"},{"tag":"v5.0.0"}]}`
 			r := io.NopCloser(strings.NewReader(json))
 			return &http.Response{
 				Body:       r,
 				StatusCode: http.StatusOK,
 			}, nil
 		}
 
-		res, err := checkForUpgrades("4.7.3", backoff)
+		res, err := checkForUpgrades(discardLogs, "4.7.3", backoff,
+			fakeClient, ctx, cfg, false)
 		assert.Equal(t, counter, 2)
 		assert.NilError(t, err)
-		assert.Equal(t, res, `{"display_name":"PGO 5.0.3","major_version":5}`)
+		assert.Equal(t, res, `{"pgo_versions":[{"tag":"v5.0.3"},{"tag":"v5.0.2"},{"tag":"v5.0.1"},{"tag":"v5.0.0"}]}`)
 	})
 }
 
 func TestCheckForUpgradesScheduler(t *testing.T) {
+	fakeClient := setupFakeClientWithPGOScheme(t, false)
```

Comment:
:thinking: Can we use `envtest` here instead of `fake`?  It is specifically the `CacheWithWait` interface that I'm thinking about, i.e. if we use `envtest` and start the manager, can we just then grab/use the client as we do in our other tests?  With this approach the cache should always sync before we grab the client for use in our tests, so wondering if that would allow us to avoid the need to create the interface simply to facilitate mocking.

Maybe this is something we can consider for a future commit (switch to `envtest`)?  I definitely don't want to blow up the current tests at this point or anything (since there are a lot of great tests in here), so lets be sure to discuss before making any changes.  Also, if anything that I'm missing, just let me know.

Reply:
I made a note to change this in the future; I did a quick version of changing it over and it was mostly clean but there were some failures, so as we discussed, I'll merge in now and we can iterate at a later point.